### ISSUE TITLE
Animation download script does not handle duplicate animations

### ISSUE THREAD

**Issue by AndyMender** [Tuxemon/Tuxemon#1314]
**Is your feature request related to a problem? Please describe.**
Currently, on the [Tuxemon wiki](https://wiki.tuxemon.org/index.php?title=Category:Technique_Animation) we have multiple copies of the same animation with different sizes. The problem is that the animation download script doesn't distinguish between them, because it strips the number suffix and from what I noticed locally it gives the impression as if the new animations have more frames.

**Describe the solution you'd like**
We have a couple of options
- Only download the first encountered animation
- Keep the suffix and treat animation variants of different dimensions as separate animations

**Describe alternatives you've considered**
Above

**Additional context**
N/A

**Comment by kerizane** [comment]
There are a couple of options.

Including the number within the file would work, but would need you to update the json files again. 

64 x 64 is the preferred size, I think we should try to use this where possible. For a few of these I assume the other size was the original, and the 64 x 64 is an edited version. 

I think we should clean the wiki up a bit, as some of these animations aren't really used - options:
1. Mark the animation as unused in the wiki (Add a category?).
2. Have a list of skipped animations in the script.
3. Delete / move the larger versions within the wiki.
4. Only download a non-standard size if 64 x 64 isn't present.

**Comment by AndyMender** [comment]
1. Sounds good. We can scrape that label via the animation download script.
2. Yup!
3. Delete/move as in from the git tree or in the wiki?
4. That should be easy to check. We cache the GIF locally in the anim download script so we can consider the 64x64 GIF to be the default (name of animation without any suffixes, etc.) and only use the differently sized GIF if no default is present

Who has access to the wiki to do the clean up by the way?

**Comment by kerizane** [comment]
@Sanglorian

**Comment by kerizane** [comment]
Up to you but it would go with option 2 or 4, as this doesn't depend on sanglorian. If we ever need to support multiple file sizes we can just change the script.

Option 2 is a bit manual, but it's the simplest, and gives you the most control. In this case you could detect that there are multiple sizes and require you to config all but one of them as being skipped? Could combine it with option 4 and only require config if there are multiple nonstandard sizes.

**Comment by AndyMender** [comment]
I'll go for option 4. Option 2 can be implemented as a log line in the script's output :)

**Comment by kerizane** [comment]
In option 2 I meant explicitly choose animations to skip as static config within the script.

**Comment by Sanglorian** [comment]
I've been through the animations and it looks like lightning claw is the only one where the larger animation is meant to be used, so I've renamed that one so it reads as a unique animation. Otherwise, the approach of taking the 64x64 animation works for me (or the smaller in the case of rockfall).

**Comment by kerizane** [comment]
Actually, public users can edit the wiki. I think we should avoid complicating the script, and just modify the wiki.

Somebody should change the category of the duplicate animations to "Unused Technique Animations" @Sanglorian any objections?

**Comment by AndyMender** [comment]
> Actually, public users can edit the wiki. I think we should avoid complicating the script, and just modify the wiki.

For sure. If we have easy access to the structure of the anim resources in the wiki, it should be possible to fix the outliers which would otherwise require tons of extra work. Another candidate mentioned in the linked PR would be `firelion_right` which currently has a 64x61 animation, even though it could easily be a 64x64 one.

**Comment by kerizane** [comment]
Yeah, I want us to fix as many of the animations as possible. Some need to be scaled, some cropped.

You can ask in the discord, but not sure how easy it will be. I think fixing the animations will need to be fairly manual anyway.

**Comment by kerizane** [comment]
Plan:

All animations on the wiki should have the Category "Technique_Animation" or "Unused_Technique_Animation". "Unused_Technique_Animation" will be for the original animations (for reference and for editing) and draft animations that we're not sure about using yet. This will make it obvious which animations will / won't be picked up by the script. 

The script should stop modifying the animation name, instead the names should be modified on the wiki. There probably won't be many that will need to be amended, as most of the ones with numbers in the name are unused (so these ones can keep the numbers in the name). 

If we do the above then we won't need to worry about duplicate animations.

I'd like to wait for feedback from Sanglorian as I think he is the main person to look after the wiki, but if he doesn't reply by Monday then feel free to get started.

**Comment by Sanglorian** [comment]
I would rather keep all animations in the same place, whether they're currently used or not -- but what about adding an separate category, "Technique Animation in Use"? Then we still have the Technique Animation category for all techniques, but the scraper can target the new category.

**Comment by kerizane** [comment]
Good point, fine with me.

Done - [[Category:Used Technique Animation]]

**Comment by AndyMender** [comment]
> The script should stop modifying the animation name, instead the names should be modified on the wiki. There probably won't be many that will need to be amended, as most of the ones with numbers in the name are unused (so these ones can keep the numbers in the name).
> 
> If we do the above then we won't need to worry about duplicate animations.

If I understand correctly, you don't want the download animation script to strip the dimension suffix from the anim name? So for instance, `rockfall_193` and `rockfall_264` should still be different animations?
I think that's fine, but then we would need the following
- have a non-suffixed animation (preferably 64x64 if possible?)
OR
- modify the technique JSON definitions to pick up the correct anims

 I would suggest the frame number to always be in the format `*_xxx` instead of the current `*xxx` so that the frame index doesn't get mixed up with anim dimensions.

> Good point, fine with me.
> 
> Done - [[Category:Used Technique Animation]]

I see it as a separate URL in the wiki, thanks!

Let's do the following then?
- switch URL to `Category:Used_Technique_Animation`
- remove stripping of anim suffix to deduplicate + fix frame indexing

**Comment by kerizane** [comment]
Sounds good!

**Comment by AndyMender** [comment]
All PR requests were merged, closing.
